### PR TITLE
[WIP, Waiting] Block some hexes on "ship to ship" battlefield

### PR DESCRIPTION
**Pull request by dydzio0614**
TODO:
Change serialization version, backwards compatibility

Waiting for FeniksFire obstacle branch and Alex CStack branch before continuing

### CODE REVIEW

**Comment by FeniksFire** [comment]
As for me it's be nice to remake everything for better concept than now but if someone don't want do it the best thing it's stick to the original concept without mixing things. 
```
//Accessibility is property of hex in battle. It doesn't depend on stack, side's perspective and so on.
enum class EAccessibility
{
...
    UNAVAILABLE, //indestructible wall parts, special battlefields (like boat-to-boat)
...
};
```

File: lib/battle/CObstacleInstance.cpp [hunk 1]
```diff
@@ -142,6 +142,19 @@ void SpellCreatedObstacle::battleTurnPassed()
 		turnsRemaining--;
 }
 
+std::vector<BattleHex> BattlefieldLimiterObstacle::getAffectedTiles() const
+{
+	std::vector<BattleHex> impassableTiles;
```
Comment (alexvins) [inline lib/battle/CObstacleInstance.cpp]: This method does not make sense. You should make BattlefieldLimiterObstacle 1-tile and put your obstacles to positions during battle setup.

File: lib/registerTypes/RegisterTypes.h [hunk 2]
```diff
@@ -201,6 +201,7 @@ void registerTypesMapObjects2(Serializer &s)
 	//s.template registerType<CObstacleInstance>();
 		s.template registerType<CObstacleInstance, MoatObstacle>();
 		s.template registerType<CObstacleInstance, SpellCreatedObstacle>();
+		s.template registerType<CObstacleInstance, BattlefieldLimiterObstacle>();
```
Comment (alexvins) [inline lib/registerTypes/RegisterTypes.h]: this line completely breaks save compatibility.

Comment (dydzio0614) [inline lib/registerTypes/RegisterTypes.h]: Yep - what is the current solution for such things? No register new classes at all?

**Comment by alexvins** [comment]
[Not a part of review] IMO, it is better to alter AccessibilityInfo calculation rather than defining new obstacle type.

**Comment by dydzio0614** [comment]
Yep, I will probably remake this branch from scratch - though I am still interested in answer to how to handle serialization when planning to register new type.

**Comment by alexvins** [comment]
New entity types must be added last (after others, before netpacks).

File: lib/registerTypes/RegisterTypes.h (hunk 2 above)
Comment (alexvins) [inline lib/registerTypes/RegisterTypes.h]: you should register between registerTypesMapObjectTypes and registerTypesClientPacks1, may add new method for current additions and use it for new types until we decide to break save compatibility.

**Pull request closed without merging**